### ISSUE TITLE
Compilation guide

### ISSUE THREAD

**Issue by yagamidev** [trusttoken/true-currencies#16]
HI,
The readme is not describing how to compile. When I'm trying it I got some errors.

Thanks.

**Comment by zachlysobey** [comment]
I was able to successfully compile using `truffle compile`.

I agree though -- it'd be nice to have more explicit instructions in the README.

My suggestion:

- break "testing" section into "installing dependencies" and "testing"
- add a new section: "building"
   - maybe suggest global `truffle` install?
   - maybe add *npm script* so we can just say *npm run build*?
      - this would point at `./node_modules/.bin/truffle build`
- consider "deploying" as a new section in the same effort?
  - I suspect this will be somewhat more difficult -- updating migrations, etc.. 

I'd actually be willing to open a PR, but I'd like to get some feedback on the above, or instructions otherwise before I take the time to do that.

**Comment by yagamidev** [comment]
Hello @zachlysobey that will be very great if you can make such PR.
I think steps you listed are right what people need to have the code setup in their working station. 
Thanks in advance

**Comment by yagamidev** [comment]
Almost the deploy script , I don't why but my truffle is not supporting the ES6 syntax.  Async , await and arrow functions. 
And also , when I remove them I end with an error saying that Gas out of Limit that's when deploying TrueUSD.Sol and timeLockedConttollrrler

**Comment by zachlysobey** [comment]
@yagamidev
are you running truffle from `node_modules` or from a global install? Make sure the version is the same. Also, I'd check your `node` & `npm` versions. Not sure if there are any explicit requirements here, but that sounds like either you need a newer version of node or truffle.

I think I'll wait to get feedback from a contributer or owner here before opening a PR for this -- don't wanna go through the effort it its the wrong approach. Also, I have a few pending PR's already :) don't wanna overwhelm 'em.

**Comment by yagamidev** [comment]
@zachlysobey I'm running it from global install. Which version should be the same?

Yes I can see you effectively have pending PRs, but can you work on that one inside your fork so that I can check before Contributor/Owner checks?
THanks

**Comment by zachlysobey** [comment]
@yagamidev perhaps I can give you a hand off-line so to speak, and we can come back and post anything we learn here. feel free to drop me a line to my gmail: zlysobey@...

But... the `package.json` depends on a version of truffle, so when you `npm install` it will be downloaded into `node_modules`. You can run this *local* `truffle` install by doing `./node_modules/.bin/truffle <cmd>` or you can just ensure your global install has the same version as the one that `npm install` downloads (4.1.8).

**Comment by kunalbarchha** [comment]
running in to an issue while using truffle compile. 

I am not using global truffle installation. Navigated to node_modules/.bin/truffle compile

Error: 
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3

**Comment by zachlysobey** [comment]
@kunalbarchha, I'm using similar node/npm versions and the node_modules/.bin/truffle script as well, and I'm not getting that error. It seems like a really weird thing to happen actually.

I wonder if its possible you've installed your dependencies using a different node version then the one you're running? Or something get changed somewhere? Does this happen with a fresh `git clone` and a fresh `npm install`?

**Comment by kunalbarchha** [comment]
@zachlysobey Tried with sudo rm -rf <trueUSD> 
cloned again. 

Then submodule init, and update. 
after that, I made sure I was on correct node version, so used NVM use 10.10.0 
but then also resulted in same error. 

So far my understanding is that, there is some conflict with my nodejs setup. I am trying to resolve it, or else I'll try on a different system with fresh ubuntu installation and see if it's just my machine or some issue with compiling itself.

**Comment by zachlysobey** [comment]
@kunalbarchha just to be absolutely sure, I repeated those steps again myself (I'm on OSX), and it *does* compile fine for me.

```sh
$ node -v
v10.11.0
$ npm -v
6.4.1
```

**Comment by wjmelements** [comment]
Which operating system are you using?

**Comment by vanruch** [comment]
Solidity contracts compilation should be straightforward now by running `yarn build:sol`.
Vyper setup can be tricky, but after #255 it should compile out of the box if there is a Docker daemon running.
